src/server/container_utils.cc

```diff
@@ -283,7 +283,15 @@ OpResult<string> RunCbOnFirstNonEmptyBlocking(Transaction* trans, int req_obj_ty
   auto wcb = [](Transaction* t, EngineShard* shard) { return t->GetShardArgs(shard->shard_id()); };
 
   *block_flag = true;
-  auto status = trans->WaitOnWatch(limit_tp, std::move(wcb));
+  const auto& keyChecker = [](EngineShard* owner, const DbContext& context, Transaction*,
+                              std::string_view key) -> bool {
+    auto [it, exp_it] = owner->db_slice().FindReadOnly(context, key);
+    (void)exp_it;
+    // Only LIST, ZSET and STREAM are allowed to block.
+    return IsValid(it) && (it->second.ObjType() == OBJ_LIST || it->second.ObjType() == OBJ_ZSET ||
```

Comment:
` (it->second.ObjType() == OBJ_LIST || it->second.ObjType() == OBJ_ZSET || it->second.ObjType() == OBJ_STREAM)` does not add much since only those types notify in the first place. this condition is always true, imho

Reply:
it will be fixed